File: docs/lib/page.js
/**
 * @template T
 */
class Page {

    /** @type {T[]} */
    items;

    /** @type {number} */
    pageIndex;

    /** @type {number} */
    pageCount;

    /**
     * @param {Page} json 
     */
    constructor(json) {
        this.items = json?.items ?? [];
        this.pageIndex = json?.pageIndex ?? 0;
        this.pageCount = json?.pageCount ?? 0;
    }

}

export {
    Page
}

File: Core/Page.cs
using KolibSoft.Catalogue.Core.Abstractions;

namespace KolibSoft.Catalogue.Core;

public class Page<T> : IPage<T>
{
    public T[] Items { get; init; } = Array.Empty<T>();
    public int PageIndex { get; init; } = 0;
    public int PageCount { get; init; } = 0;
}